Small mathematical difference when using np.log

https://github.com/pkeilbach/htwg-practical-nlp/blob/e1e227156bca53cdc0da2b963eacf44e779613c9/tests/htwgnlp/test_naive_bayes.py#L111

There is a small mathematical difference between `np.log(6/2)` and `np.log(6) - np.log(2)`.

So you can change the assertion to 
`np.testing.assert_allclose(model.logprior, expected)`
to get rid of the problem

It is the same for the function `def _get_log_ratios(self)`
There is a slight difference between / and -
With / it passed but with - it throws `AssertionError: assert 0.6931471805599454 == 0.6931471805599453`

--- a/tests/htwgnlp/test_naive_bayes.py
+++ b/tests/htwgnlp/test_naive_bayes.py
@@ -4,7 +4,7 @@ import numpy as np
 import pandas as pd
 import pytest
 
-from htwgnlp.naive_bayes import NaiveBayes
+from htwgnlp.naive_bayes_private import NaiveBayes
 
 model = NaiveBayes()
 
@@ -107,7 +107,8 @@ def trained_log_ratios(train_samples, train_samples_labels):
 )
 def test_set_logprior(labels, expected):
     model.logprior = labels
-    assert model.logprior == expected
+    np.testing.assert_allclose(model.logprior, expected)
+    # assert model.logprior == expected
 
 
 @pytest.mark.parametrize(
@@ -171,15 +172,15 @@ def test_get_log_ratios(trained_log_ratios):
     assert isinstance(trained_log_ratios, pd.Series)
     assert trained_log_ratios.index.size == 9
 
-    assert trained_log_ratios.loc["happy"] == np.log(0.15 / 0.05)
-    assert trained_log_ratios.loc["sad"] == np.log(0.05 / 0.15)
-    assert trained_log_ratios.loc["love"] == np.log(0.1 / 0.05)
-    assert trained_log_ratios.loc["hate"] == np.log(0.05 / 0.1)
-    assert trained_log_ratios.loc["weather"] == 0.0
-    assert trained_log_ratios.loc["I"] == 0.0
-    assert trained_log_ratios.loc["am"] == 0.0
-    assert trained_log_ratios.loc["because"] == 0.0
-    assert trained_log_ratios.loc["the"] == 0.0
+    np.testing.assert_allclose(trained_log_ratios.loc["happy"], np.log(0.15 / 0.05))
+    np.testing.assert_allclose(trained_log_ratios.loc["sad"], np.log(0.05 / 0.15))
+    np.testing.assert_allclose(trained_log_ratios.loc["love"], np.log(0.1 / 0.05))
+    np.testing.assert_allclose(trained_log_ratios.loc["hate"], np.log(0.05 / 0.1))
+    np.testing.assert_allclose(trained_log_ratios.loc["weather"], 0.0)
+    np.testing.assert_allclose(trained_log_ratios.loc["I"], 0.0)
+    np.testing.assert_allclose(trained_log_ratios.loc["am"], 0.0)
+    np.testing.assert_allclose(trained_log_ratios.loc["because"], 0.0)
+    np.testing.assert_allclose(trained_log_ratios.loc["the"], 0.0)
 
 
 @pytest.mark.parametrize(

